front/src/applications/editor/tools/trackEdition/utils.ts

```diff
@@ -29,3 +30,15 @@ export function injectGeometry(track: EditorEntity): EditorEntity {
     },
   };
 }
+
+/**
+ * Remove the invalid ranges when the length of the track section has been modified
+ * - keep ranges if begin is undefined in case we just added a new one or if we deleted the begin input value
+ * - remove ranges which start after the new end
+ * - cut the ranges which start before the new end but end after it
+ */
+export function removeInvalidRanges<T>(values: LinearMetadataItem<T>[], newLength: number) {
+  return values
+    .filter((item) => item.begin < newLength || !item.begin)
```

Comment:
Why did you add a condition `|| !item.begin` here ?
```suggestion
    .filter((item) => item.begin < newLength)
```